File: sample/sphere_sample/src/main/java/com/sphere/sample/SampleWebViewActivity.java
package com.sphere.sample;

import android.app.Activity;
import android.os.Bundle;
import android.webkit.WebView;

import com.sphere.analytics.SphereJsInterface;

public class SampleWebViewActivity extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_webview);

        // Initialize the webview
        WebView webView = findViewById(R.id.webview);
        webView.getSettings().setJavaScriptEnabled(true);


        // Add Sphere JavaScript Interface for Sphere Analytics
        webView.addJavascriptInterface(new SphereJsInterface(), SphereJsInterface.getHandlerName());


        // Navigate to site
        webView.loadUrl("file:///android_asset/index.html");
//        webView.loadUrl("your website url");
    }
}

File: sample/sphere_sample_kt/src/main/java/com/sphere/sample/kt/SampleWebViewActivity.kt
package com.sphere.sample.kt

import android.app.Activity
import android.os.Bundle
import android.util.Log
import android.webkit.JavascriptInterface
import android.webkit.WebView
import com.sphere.analytics.SphereJsInterface
import org.json.JSONObject

class SampleWebViewActivity : Activity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_webview)

        // Initialize the webview
        val webView = findViewById<WebView>(R.id.webview)
        webView.settings.javaScriptEnabled = true


        // Add Sphere JavaScript Interface for Sphere Analytics
        webView.addJavascriptInterface(TagJSInterface(), "TagJSInterface")


        // Navigate to site
        webView.loadUrl("file:///android_asset/index.html")
//        webView.loadUrl("your website url");
    }
}

class TagJSInterface {
    @JavascriptInterface
    fun postMessage(message: String?) {
        if(message == null)  return
        try {
            val jsonObject = JSONObject(message)
            when (jsonObject.optString("name")) {

                SphereJsInterface.getHandlerName() -> {
                    SphereJsInterface.handlePostMessageJsonObject(jsonObject)
                }

            }
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }
}